go.mod

```diff
@@ -43,18 +43,19 @@ require (
 	golang.org/x/mod v0.14.0
 	golang.org/x/text v0.14.0
 	gopkg.in/yaml.v3 v3.0.1
-	k8s.io/api v0.27.2
-	k8s.io/apimachinery v0.27.2
-	k8s.io/client-go v0.27.2
+	k8s.io/api v0.28.3
+	k8s.io/apimachinery v0.28.3
+	k8s.io/apiserver v0.28.3
+	k8s.io/client-go v0.28.3
 	k8s.io/cluster-bootstrap v0.27.2
-	k8s.io/component-base v0.27.2
-	k8s.io/klog/v2 v2.90.1
+	k8s.io/component-base v0.28.3
+	k8s.io/klog/v2 v2.110.1
 	k8s.io/kubectl v0.27.2
 	k8s.io/utils v0.0.0-20230726121419-3b25d923346b
 	sigs.k8s.io/cloud-provider-azure v1.27.7
 	sigs.k8s.io/cluster-api v1.5.3
 	sigs.k8s.io/cluster-api/test v1.5.3
-	sigs.k8s.io/controller-runtime v0.15.1
+	sigs.k8s.io/controller-runtime v0.16.3
```

Comment:
is it ok to make this change before bumping CAPI to 1.6?

Reply:
I believe this is a ASO v2.4.0 requirement.

I don't think there would be a strict requirement of staying in sync w/ CAPI (though certainly a best practice). If there were k8s breaking API ramifications our E2E tests would tell us that. Do we have a documented contract to stick to the same version as CAPI's latest version in CAPZ (and other providers)?

Reply:
I think you're right

let's ask in the CAPI office hours tomorrow am to get confirmation?

Reply:
Confirmed in CAPI this is not a strict requirement